spec/features/openid_connect/openid_connect_spec.rb

```diff
@@ -119,6 +119,44 @@
         to be_a NullServiceProviderRequest
       expect(page.get_rack_session.keys).to_not include('sp')
     end
+
+    it 'auto-allows and sets the correct CSP headers after an incorrect OTP' do
+      client_id = 'urn:gov:gsa:openidconnect:sp:server'
+      user = user_with_2fa
+
+      IdentityLinker.new(user, client_id).link_identity
```

Comment:
Why is this necessary? Doesn't the controller + form already do this? In a feature spec, I would expect to only perform actions as a user would.